WordPress/gutenberg
# Replace "Add new post" link text with more meaningful Label

## What?

This PR replaces the static "*Add new post*" link text inside the the query block, with a more meaningful label from the currently selected post type.

## Why?

This PR is [polishing the existing UI](https://github.com/WordPress/gutenberg/pull/27732) with data, that is already available, without introducing any new elements. 

## How?

Removes a static text string, with the `add_new_item` label from the currently selected post type.

## Testing Instructions


1. Open a post or page.
2. Insert a query block.
3. Open the inspector controls
4. Switch the queried post type
5. And see the link text change, depending on the selected post type, within the `CreateNewPostLink` element.

### Testing Instructions for Keyboard

*Test can be done by keyboard as well, without any additions to my **Testing Instruction**.*

## Screenshots or screencast 

### Before

https://github.com/WordPress/gutenberg/assets/198883/a2bdb869-305a-448d-b47c-7a201db8dfdb

### After

https://github.com/WordPress/gutenberg/assets/198883/593ee111-2674-400c-9bac-420b68fa4578
